src/AppInstallerCLICore/Workflows/PortableInstallFlow.cpp

```diff
@@ -0,0 +1,376 @@
+
+            Key key;
+            if (scope == Manifest::ScopeEnum::Machine)
+            {
+                key = Registry::Key::Create(HKEY_LOCAL_MACHINE, std::wstring{ s_PathSubkey_Machine });
+            }
+            else
+            {
+                key = Registry::Key::Create(HKEY_CURRENT_USER, std::wstring{ s_PathSubkey_User });
+            }
+
+            std::wstring pathName = std::wstring{ s_PathName };
+            std::string portableLinksDir = Normalize(linksDirectory.u8string());
+            std::string pathValue = Normalize(key[pathName]->GetValue<Value::Type::String>());
+            
+            if (pathValue.find(portableLinksDir) == std::string::npos)
+            {
+                if (pathValue.back() != ';')
+                {
+                    pathValue += ";";
+                }
+
+                pathValue += portableLinksDir + ";";
+                AICLI_LOG(CLI, Info, << "Adding to Path environment variable: " << portableLinksDir);
+                key.SetValue(pathName, ConvertToUTF16(pathValue), REG_EXPAND_SZ);
+                return true;
+            }
+            else
+            {
+                AICLI_LOG(CLI, Verbose, << "Path already existed in environment variable. Skipping...");
+                return false;
+            }
+        }
+
+        void WritePortableEntryToUninstallRegistry(Execution::Context& context)
+        {
+            const AppInstaller::Manifest::Manifest& manifest = context.Get<Execution::Data::Manifest>();
+            const Manifest::AppsAndFeaturesEntry& entry = GetAppsAndFeaturesEntryForPortableInstall(context.Get<Execution::Data::Installer>()->AppsAndFeaturesEntries, manifest);
+            const std::string& packageIdentifier = manifest.Id;
+
+            std::string sourceIdentifier;
+            if (context.Contains(Execution::Data::Source))
+            {
+                sourceIdentifier = context.Get<Execution::Data::Source>().GetIdentifier();
+            }
+            else
+            {
+                sourceIdentifier = s_LocalSource;
+            }
+
+            const std::wstring& productCode = ConvertToUTF16(packageIdentifier + "_" + sourceIdentifier);
+
+            Portable::PortableARPEntry uninstallEntry = Portable::PortableARPEntry(
+                ConvertToScopeEnum(context.Args.GetArg(Execution::Args::Type::InstallScope)),
+                context.Get<Execution::Data::Installer>()->Arch,
+                productCode);
+
+            if(uninstallEntry.Exists())
+            {
+                if (uninstallEntry.IsSamePortablePackageEntry(packageIdentifier, sourceIdentifier))
+                {
+                    // TODO: Replace HashOverride with --Force when argument behavior gets updated.
+                    if (!context.Args.Contains(Execution::Args::Type::HashOverride))
+                    {
+                        AICLI_LOG(CLI, Error, << "Registry match failed, skipping write to uninstall registry");
```

Comment:
Is there a test case which covers this upgrade scenario?

Reply:
Not covering upgrade scenario in this PR, will be in the next one.